src/sentry/static/sentry/app/views/performance/traceDetails/transactionBar.tsx

```diff
@@ -223,11 +231,47 @@ class TransactionBar extends React.Component<Props> {
     );
   }
 
+  renderRectangle() {
+    const {transaction, traceInfo, theme} = this.props;
+
+    const palette = theme.charts.getColorPalette(traceInfo.maxGeneration);
+
+    // TODO(tonyx) Does this need to handle any unexpected data? (e.g. start > end)
+    const delta = traceInfo.endTimestamp - traceInfo.startTimestamp;
+    const startPercentage =
+      (transaction.start_timestamp - traceInfo.startTimestamp) / delta;
+    const duration = transaction.timestamp - transaction.start_timestamp;
+    const widthPercentage = duration / delta;
```

Comment:
What if delta is 0?

Reply:
I think this belongs in the same category as the earlier comment about bad timestamps and might be something the backend should handle.

I will note that the span view does explicitly handle these cases, I simply chose to leave it out in this iteration for the sake of simplicity.